tests/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+#![no_std]
+#![feature(in_band_lifetimes)]
+
+extern crate capsules;
+#[macro_use] extern crate kernel;
+
+#[cfg(feature = "sam4l-test")]
```

Comment:
Where do these `sam4l-test` style features come from / get defined?

Reply:
Formally, these are `defined` ofc in this crate (the test crate) but these features are initialized from each board crate as optional dependencies and are only enabled when you enable the `test` feature in your `board crate`!

So for example, you ```make imix test```` it will:
* compile test crate
* compile a special version of the reset handler
* run the tests and `panics`